app/models/throttle.rb

```diff
@@ -48,10 +49,51 @@ class Throttle < ApplicationRecord
     },
   }.freeze
 
+  # @param [User,Integer,String] target User, its ID, or a string identifier
+  # @param [Symbol] throttle_type
+  # @return [Throttle]
+  def self.for(target:, throttle_type:)
```

Comment:
What do you think of having the caller of this method explicitly specify whether they want to use the user_id target or the string target to narrow the scope? My worry is a future user of the function is passing in an argument they want to be treated as a user_id, but it is a string for some reason and ends up using the wrong column.

My feeling would be to use `find_or_create_by(user_id:, throttle_type:)` or `find_or_create_by(target:, throttle_type:)` directly, but do understand wanting a common interface, so perhaps there's a way to handle it with parameters to the existing function.

Reply:
how about a signature like this:

```suggestion
  def self.for(user: nil, user_id: nil, target: nil, throttle_type:)
```

so you can call with
- `.for(user: user)` 
- `.for(user_id: user_id)`
- `.for(target: target)`

and then we throw an error if those are all nil

Reply:
yep, that makes sense to me, though I would cut `user:` 🙂

Reply:
I was gonna go the opposite way and drop `user_id` so we minimize the chances of orphaned rows

Reply:
Ah right, I wasn't sure if a full user was available in all cases 👍🏼

Reply:
Updated in 509a4e6968623894713eac7161139ef0a7765b68